### PR TITLE
update dependabot to enable CVE detect

### PR DESCRIPTION
**Pull request by charles-chenzz**
partially addresses #5259

### CODE REVIEW

File: .github/dependabot.yml
Comment (charles-chenzz) [inline .github/dependabot.yml:12]: limit to 10 can avoid bot sending too much PRs (which not very often I think but just limit here

File: .github/dependabot.yml
Comment (charles-chenzz) [inline .github/dependabot.yml:14]: I wasn't 100% sure this field will work. the safest way is refs to https://docs.github.com/en/code-security/dependabot/dependabot-alerts/configuring-dependabot-alerts
(need to configure by maintainer or user who have permission to write of this repo

**Comment by natasha41575** [comment]
@antoooks would you be able to take a look at this one?

File: .github/dependabot.yml
Comment (natasha41575) [reply .github/dependabot.yml:14]: Do you know how does it tell us about the CVE detection? 

I feel like we won't want the CVEs to be public knowledge until we can fix and release them, so I want to be sure that dependabot won't start opening public PRs/issues about CVEs if we enable this. If it will, then we probably shouldn't add it and will have to come up with another way to do CVE detection.

Comment (charles-chenzz) [reply .github/dependabot.yml:14]: I need to investigate about it. will share on slack when I get more context

Comment (charles-chenzz) [reply .github/dependabot.yml:14]: image

I use my personal repo and manually enable CVE detect, it seems like it will show up here if there's any CVEs. but I'm not quite sure if it's visible for non-maintainer.

Comment (natasha41575) [reply .github/dependabot.yml:14]: Screenshot 2023-10-26 at 4 33 46 PM

Looks like it is only visible to maintainers, so that's great! Thank you for looking into it

Comment (charles-chenzz) [reply .github/dependabot.yml:14]: can you review other part of this PR? I think it's cover all we need currently(but still use simple method, if we need more complex logic then we need to see)

Comment (natasha41575) [reply .github/dependabot.yml:14]: Will add this to my review list for next week!

Comment (natasha41575) [reply .github/dependabot.yml:14]: Just took a quick look. I think the rest of this PR LGTM, but I think another eye will be good. @antoooks or @koba1t would one of you be able to take a look too?

**Review by natasha41575 — APPROVED** [review]

**Comment by sarab97** [comment]
I would suggest considering adding reviewer in the config to auto assign. More often than not such PRs goes unnoticed.
Utlizing option `reviewers` and `assignees` can help with that.
Also, adding some labels by default. Like `ok-to-test` is basic one that can be considered. And maybe we can go another step in using `lgtm` by default.
I have noticed many repos use labels to auto approve and close such PRs without the need of any interaction from any maintainers.

**Comment by charles-chenzz** [comment]
> I would suggest considering adding reviewer in the config to auto assign.

prow will auto-assign reviewer so I think we don't need to re-do it again. but we can consider let specify reviewers to review.

> And maybe we can go another step in using lgtm by default.
I have noticed many repos use labels to auto approve and close such PRs without the need of any interaction from any maintainers.

don't think it's a good idea. as some deps might have breaking changes and auto approve will introduce troubles.

**Comment by koba1t** [comment]
@charles-chenzz 

Thanks for your great contribution!!!

So, Looks lie I think we need clearly one question.

> We need to be careful though, to ensure that we are not bumping any dependencies past the dependencies in kubernetes/kubernetes.

I found noticed the above problem on the https://github.com/kubernetes-sigs/kustomize/issues/5259.
Did you make any guardrails for this?
If this is my misunderstanding, I apologize for my mistake.

**Comment by charles-chenzz** [comment]
@koba1t hi, this PR target at CVE detect. I think that's the main purpose of the issue. also, I'm investigating how to manage deps in multi git repos

**Comment by koba1t** [comment]
@charles-chenzz 
> this PR target at CVE detect. I think that's the main purpose of the issue. also, I'm investigating how to manage deps in multi git repos

Thanks for describing!
I think may PR title is missing, Could you fix that?

**Comment by charles-chenzz** [comment]
@koba1t changed. revisit when you get a chance and see if it make sense to you. Thanks!

**Comment by sarab97** [comment]
> don't think it's a good idea. as some deps might have breaking changes and auto approve will introduce troubles.

I agree on not to auto merge without anyone's review. But make it easy enough to just review once and make tests auto run on dependabot PRs. However this can be taken up seperately.


> So, Looks like I think we need clearly one question.
> 
> > We need to be careful though, to ensure that we are not bumping any dependencies past the dependencies in kubernetes/kubernetes.
> 
> I found noticed the above problem on the #5259. Did you make any guardrails for this?

@koba1t I dont believe there is any native way in dependabot to track version updates like that in another project. 
We just have to manually verify that.

**Comment by natasha41575** [comment]
> I would suggest considering adding reviewer in the config to auto assign. More often than not such PRs goes unnoticed.
Utlizing option reviewers and assignees can help with that.

I think we already get auto-assigned reviewers from the k8s bot so I don't think we need to add this to the dependabot config necessarily. Please feel free to double check and correct me if that doesn't happen today.

> Also, adding some labels by default. Like ok-to-test is basic one that can be considered. And maybe we can go another step in using lgtm by default.

I think this is a good idea, but as discussed offline we can file an issue and tackle this separately from this PR. PRs probably shouldn't auto-merge without anyone looking at them (so we can check for breaking changes, and restrictions from the k/k repo, etc); I think as part of this we (the kustomize maintainers) need to publish some sort of guidance for how to review PRs that just update dependencies. Adding LGTM and ok-to-test should be fine as the PRs won't merge automatically without approval, so I think this can help reduce the maintenance burden without causing immediate issues.

Thank you @sarab97 for the suggestions - I'm glad we are proactively thinking about how we can reduce maintenance and PR review burden.

**Comment by natasha41575** [comment]
/hold cancel

Thank you @charles-chenzz!

**Comment by charles-chenzz** [comment]
@antoooks hi, I would file another PR to adjust the changes. Thanks!

**Pull request merged**
